app/src/main/java/org/fossasia/pslab/communication/sensors/Sx1276.java

```diff
@@ -0,0 +1,454 @@
+        // Output Power 17dbm
+        setTxPower(17, PA_OUTPUT_RFO_PIN);
+        idle();
+        // Set bandwidth
+        setSignalBandwidth(125e3);
+        setSpreadingFactor(12);
+        setCodingRate4(5);
+    }
+
+    private void initiateSX1276(SPI spi, int frequency) throws IOException {
+
+        this.spi = spi;
+        this.spi.setParameters(2, 6, 1, 0, 1);
+        this.frequency = frequency;
+        this.name = "SX1276";
+
+        reset();
+
+        this.version = SPIRead(REG_VERSION, 1).get(0);
+        if (version != 0x12) {
+            Log.d(name, "Version error " + version);
+        }
+        sleep();
+        setFrequency(frequency);
+        // Set base address
+        setupBytesArray(0);
+        SPIWrite(REG_FIFO_TX_BASE_ADDR, bytes);
+        SPIWrite(REG_FIFO_RX_BASE_ADDR, bytes);
+        // Set LNA boost
+        setupBytesArray(SPIRead(REG_LNA, 1).get(0) | 0x03);
+        SPIWrite(REG_LNA, bytes);
+        // Set auto ADC
+        setupBytesArray(0x04);
+        SPIWrite(REG_MODEM_CONFIG_3, bytes);
+    }
+
+    private void reset() {
+        /**/
+    }
+
+    private ArrayList<Byte> SPIRead(int adr, int total_bytes) throws IOException {
+        // CS1 => 9
+        ArrayList<Byte> data = new ArrayList<>();
+        data.add((byte) adr);
+        for (int i = 0; i < total_bytes; i++) {
+            data.add((byte) 0);
+        }
+        data.remove(0);
+        return spi.xfer(9, data);
+    }
+
+    private void beginPacket(boolean implicitHeader) throws IOException {
+        idle();
+        if (implicitHeader) {
+            implicitHeaderMode();
+        } else {
+            explicitHeaderMode();
+        }
+        // reset FIFO & payload length
+        bytes.clear();
+        bytes.add((byte) 0);
+        SPIWrite(REG_FIFO_ADDR_PTR, bytes);
+        SPIWrite(REG_PAYLOAD_LENGTH, bytes);
+    }
+
+    private void endPacket() throws InterruptedException, IOException {
+        // put in TX mode
+        bytes.clear();
+        bytes.add((byte) (MODE_LONG_RANGE_MODE | MODE_TX));
+        SPIWrite(REG_OP_MODE, bytes);
+
+        while (true) { // Wait for TX done
+            if ((SPIRead(REG_IRQ_FLAGS, 1).get(0) & IRQ_TX_DONE_MASK) == 1) {
+                break;
+            } else {
+                Thread.sleep(100);
+            }
+        }
+        setupBytesArray(IRQ_TX_DONE_MASK);
+        SPIWrite(REG_IRQ_FLAGS, bytes);
+    }
+
+    private int parsePacket(int size) throws IOException {
+        packetLength = 0;
+        int irqFlags = SPIRead(REG_IRQ_FLAGS, 1).get(0);
+        if (size > 0) {
+            implicitHeaderMode();
+            setupBytesArray(size & 0xFF);
+            SPIWrite(REG_PAYLOAD_LENGTH, bytes);
+        } else {
+            explicitHeaderMode();
+        }
+        setupBytesArray(irqFlags);
+        SPIWrite(REG_IRQ_FLAGS, bytes);
+
+        if (((irqFlags & IRQ_RX_DONE_MASK) == 1) && ((irqFlags & IRQ_PAYLOAD_CRC_ERROR_MASK) == 1)) {
```

Comment:
(irqFlags & IRQ_PAYLOAD_CRC_ERROR_MASK) == 1) , it should be 0. Check python repo.